### ISSUE TITLE
Feature Ideas

### ISSUE THREAD

**Issue by johannesvollmer** [johannesvollmer/exr-tools#4]
Brainstorming of tool Ideas. Longer discussion  of individual ideas should probably be done in separate issues, this serves mainly as a pool of ideas.

**Comment by johannesvollmer** [comment]
Copy all attributes from one image to the other

**Comment by johannesvollmer** [comment]
Crop, Auto Crop

**Comment by Narann** [comment]
Create empty (size, color, multiview, etc)

**Comment by johannesvollmer** [comment]
Minimize: Compress Smallest + Auto Crop + f32 -> F16 + Remove Meta (optional)?

**Comment by johannesvollmer** [comment]
Convert file data between F16 <-> F32

**Comment by johannesvollmer** [comment]
Merge all layers ("multi-part image parts) into one file, separate layers of one file

**Comment by johannesvollmer** [comment]
Make compatible: multi part into separate files, tiles to scan lines, new compression methods to old methods, ...

**Comment by Narann** [comment]
Extract layers with pattern, like Guerilla: http://guerillarender.com/doc/2.3/User%20Guide_Rendering_File%20naming.html

If you have one big EXR with many layers (this can happen when using renderers) you can extract them in separated files:

```
all.exr -> %L/%L_%C.exr:

char1/char1_color.exr
char1/char1_depth.exr
```

Here, `%L` is layer name, and `%C` is channel name.

**Comment by johannesvollmer** [comment]
A file rename based on the image attributes? Interesting :)

Edit: no wait, you said extract layers, so I assume the layers are also renamed based on their attributes after extraction?

**Comment by johannesvollmer** [comment]
@Narann so, are you interested in merging some of your code?

**Comment by Narann** [comment]
Yes, I just need to find some time to put my code into `exr-tools` code base. :)

**Comment by johannesvollmer** [comment]
Also, see https://github.com/johannesvollmer/exrs/issues/108
